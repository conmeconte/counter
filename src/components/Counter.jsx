import React, { Component, Fragment } from 'react';

const styles = {
  label: {
    fontSize: 20,
    fontWeight: 'bold',
    margin: 'auto'
  },
  btn: {
    padding: 20
  }
}

class Counter extends Component {
  state = {
    count: 0
  }

  makeIncrement = amount => this.setState({ count: this.state.count + amount });

  increment = () => this.makeIncrement(1);
  decrement = () => this.makeIncrement(-1);

  render() {
    return (
      <Fragment>
        <p id='count' style={styles.label}>CLICK COUNT: {this.state.count}</p>
        <button 
          id='increment-btn' 
          onClick={this.increment}
          style={styles.btn}
        >
            ++++
        </button>
        <button 
          id='decrement-btn' 
          onClick={this.decrement}
          style={styles.btn}
        >
            DECREMENT
        </button>
      </Fragment>
    )
  }
}

export default Counter;