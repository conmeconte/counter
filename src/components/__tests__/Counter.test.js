import React from 'react';
import {shallow} from 'enzyme';
import Counter from '../Counter';
import renderer from 'react-test-renderer';

describe('Counter component layout', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('has a click count print', () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper.find('#count').exists()).toEqual(true);
  })

  it('has an increment btn', () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper.find('#increment-btn').exists()).toEqual(true);
  })

  it('has a decrement btn', () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper.find('#decrement-btn').exists()).toEqual(true);
  })
})

describe('Counter component UI', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<Counter />);
    const text = wrapper.find('#count').text();

    expect(text).toEqual('CLICK COUNT: 0');
  })

  it('increments count when increment btn is clicked', () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find('#increment-btn');
    
    incrementBtn.simulate('click');
    const text = wrapper.find('#count').text();

    expect(text).toEqual('CLICK COUNT: 1');
  })

  it('decrements count when decrement btn is clicked', () => {
    const wrapper = shallow(<Counter />);
    const decrementBtn = wrapper.find('#decrement-btn');
    
    decrementBtn.simulate('click');
    const text = wrapper.find('#count').text();

    expect(text).toEqual('CLICK COUNT: -1');
  })
})