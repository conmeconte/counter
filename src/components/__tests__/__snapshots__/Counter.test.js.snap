// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component layout matches the snapshot 1`] = `
Array [
  <p
    id="count"
    style={
        Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "margin": "auto",
          }
    }
>
    CLICK COUNT: 
    0
</p>,
  <button
    id="increment-btn"
    onClick={[Function]}
    style={
        Object {
            "padding": 20,
          }
    }
>
    ++++
</button>,
  <button
    id="decrement-btn"
    onClick={[Function]}
    style={
        Object {
            "padding": 20,
          }
    }
>
    DECREMENT
</button>,
]
`;
